app/recurring_job.go

```diff
@@ -740,12 +744,12 @@ func (job *Job) findARandomReadyNode(v *longhornclient.Volume) (string, error) {
 		return "", err
 	}
 
-	engineImage, err := job.api.EngineImage.ById(types.GetEngineImageChecksumName(v.CurrentImage))
+	engineImage, err := job.GetEngineImage(types.GetEngineImageChecksumName(v.CurrentImage))
```

Comment:
Do you mean `engineImage, err := job.api.EngineImage.ById(types.GetEngineImageChecksumName(v.CurrentImage))` also error due to decoding problem?

Reply:
Correct.